src/molinillo_solver.cr

```diff
@@ -46,6 +46,11 @@ module Shards
       packages = [] of Package
       result.each do |v|
         spec = v.payload.as(Spec) || raise "BUG: returned graph payload was not a Spec"
```

Comment:
Just noticing this: `v.payload.as(Spec)` can't possibly be falsey because it raises if the type doesn't fit. The raise branch won't be reached ever. Should this be `.as?` instead?

Reply:
Good catch!